### ISSUE TITLE
Problems with digital keys

### ISSUE THREAD

**Issue by lutaev** [jsdoc3/jsdoc#1016]
Some comments cause an error, for example this:
```javascript
/**
     * Возвращает опции для виджета
     * @returns {{legend: {position: string}, pointSize: number, height: number, lineWidth: number, fontSize: number, tooltip: {isHtml: boolean}, fontName: string, chartArea: {left: string, width: number, height: string}, series: {0: {pointSize: number, lineWidth: number, color: string}, 1: {pointSize: number, color: string, lineWidth: number, type: string, areaOpacity: number}}, vAxis: {format: string, gridlines: {color: string, count: number}, baselineColor: string, viewWindowMode: string, viewWindow: {min: number}, textStyle: {color: string}}, hAxis: {gridlines: {color: string, count: *}, baselineColor: string, textStyle: {color: string}, ticks: (Timeweb.Views.ProfitChart.chartTicks|*)}}}
     * @private
     */
_getChartOptions: function()
    {
// ... some code...
}
```
The problem is in piece of comment with digital object keys:
```javascript
// ...series: {0: {pointSize: number ... }, 1: { ... } }, ...
```
If I replace 0 and 1 by words, everything is OK.

**Comment by hegemonic** [comment]
Confirmed. This appears to be a bug in Catharsis, the library we use to parse type expressions.

**Comment by hegemonic** [comment]
Fixed on master and the releases/3.3 branch. The fix will be included in JSDoc 3.3.1.

You can get the fix now by running `npm install catharsis@0.8.7` in your JSDoc directory.
